tls/s2n_tls13_secrets.c

```diff
@@ -501,3 +501,31 @@ S2N_RESULT s2n_tls13_derive_secret(struct s2n_connection *conn, s2n_extract_secr
     RESULT_GUARD(s2n_trigger_secret_callbacks(conn, secret, secret_type, mode));
     return S2N_RESULT_OK;
 }
+
+S2N_RESULT s2n_tls13_secrets_finish(struct s2n_connection *conn)
+{
+    RESULT_ENSURE_REF(conn);
+    if (conn->actual_protocol_version < S2N_TLS13) {
```

Comment:
For background: the TLS1.2 and TLS1.3 secrets are very different, and stored separately. If I did want to wipe the TLS1.2 secrets, it would not be via this method. I've kind of hidden it behind the CONN_SECRET macro, but the secrets I'm wiping are stored at "conn->secrets.tls13.<secret_name>". If I wanted to wipe the TLS1.2 secrets, I'd need to specifically wipe them from their own storage location. I think I /could/ safely wipe them, but I'd want to do some research before I made that change, and this method would still be TLS1.3-specific.

This method only runs for TLS1.3 because all the logic in it only makes sense for a TLS1.3 connection. Let me know if you still want a comment.

Reply:
> This method only runs for TLS1.3 because all the logic in it only makes sense for a TLS1.3 connection

nope that makes sense. its in the `tls13` file which i overlooked.

Reply:
created an issue: https://github.com/aws/s2n-tls/issues/3214